Riot/Modules/Room/VoiceMessages/VoiceMessageAttachmentCacheManager.swift

```diff
@@ -105,6 +123,23 @@ class VoiceMessageAttachmentCacheManager {
         }
     }
     
+    func clearCache() {
+        for key in completionCallbacks.keys {
+            invokeFailureCallbacksForIdentifier(key.eventIdentifier, requiredNumberOfSamples: key.requiredNumberOfSamples, error: VoiceMessageAttachmentCacheManagerError.cancelled)
+        }
+        
+        operationQueue.cancelAllOperations()
```

Comment:
I notice we are calling `cancelAllOperations` here but we don't use `operation.cancelled` so not sure this will really do anything.
Is the design that we let in-flight operations complete, but then the success callback will not be called as the `invokeFailureCallbacksForIdentifier` will remove it from the callbacks?

Reply:
Good catch, this is a total miss on my part. I was under the impression that the queued operations will be removed from the queue while the in flight one would just no-op. This being a serial queue that would've been fine. I'll try to figure out how to actually cancel these block operations.

Reply:
Could maybe check 'operation.cancelled' at the end of any async work and not save files to disk, call appropriate failure callback etc? But yea whatever works for you.

Reply:
Yeah.. problem is I don't have a access to an actual operation through that `addOperation(_ block: @escaping () -> Void)` method.

Reply:
Oh weird, i thought it gave it to you. 🤔

Reply:
After a bit of testing it turns out that enqueued not running operations will be automatically cancelled after being marked as such through a `cancelAllOperations()`. I think we're okay to go ahead with the implementation as it is.

Reply:
Yea you're right, looks like you get cancellation with block operations, probably why they don't need give you access to the operation. A subclass of Operation wouldn't get that for free. So LGTM!